Source/WebCore/platform/graphics/gbm/GraphicsLayerContentsDisplayDelegateGBM.cpp

```diff
@@ -39,24 +39,20 @@ GraphicsLayerContentsDisplayDelegateGBM::GraphicsLayerContentsDisplayDelegateGBM
     : GraphicsLayerContentsDisplayDelegateTextureMapper(TextureMapperPlatformLayerProxy::create(TextureMapperPlatformLayerProxy::ContentType::WebGL))
     , m_isOpaque(isOpaque)
 {
-    m_proxy->setSwapBuffersFunction([this](TextureMapperPlatformLayerProxy& proxy) mutable {
-        if (!m_buffer)
-            return;
-
-        OptionSet<TextureMapperFlags> flags = TextureMapperFlags::ShouldFlipTexture;
-        if (!m_isOpaque)
-            flags.add(TextureMapperFlags::ShouldBlend);
-
-        proxy.pushNextBuffer(CoordinatedPlatformLayerBufferDMABuf::create(Ref { *m_buffer }, flags, WTFMove(m_fence)));
-    });
 }
 
 GraphicsLayerContentsDisplayDelegateGBM::~GraphicsLayerContentsDisplayDelegateGBM() = default;
 
 void GraphicsLayerContentsDisplayDelegateGBM::setDisplayBuffer(RefPtr<DMABufBuffer>& displayBuffer, std::unique_ptr<GLFence>&& fence)
 {
     m_buffer = displayBuffer;
-    m_fence = WTFMove(fence);
+    if (!m_buffer)
+        return;
+
+    OptionSet<TextureMapperFlags> flags = TextureMapperFlags::ShouldFlipTexture;
+    if (!m_isOpaque)
+        flags.add(TextureMapperFlags::ShouldBlend);
+    m_proxy->pushNextBuffer(CoordinatedPlatformLayerBufferDMABuf::create(Ref { *m_buffer }, flags, WTFMove(fence)));
```

Comment:
I'm not sure there's a problem, so even less sure if there's a solution, but this block here...